aviary/interface/methods_for_level2.py

```diff
@@ -564,101 +566,42 @@ def _get_2dof_phase(self, phase_name):
-                'accel': get_accel
-            }
-
-            # Set the phase function based on the phase name
-            if 'climb' in phase_name:
-                phase_functions[phase_name] = get_climb
-            elif 'desc' in phase_name:
-                phase_functions[phase_name] = get_descent
-
-            # Get the phase function corresponding to the phase name
-            phase_func = phase_functions.get(phase_name)
-
-            # Calculate the phase by calling the phase function
-            # with the transcription object and remaining phase options
-            trimmed_phase_options = {k: v for k, v in phase_options.items(
-            ) if k not in ['num_segments', 'order', 'initial_guesses', 'throttle_setting', 'external_subsystems']}
-
-            # define expected units for each phase option
-            expected_units = {
-                'alt': 'ft',
-                'mass': 'lbm',
-                'distance': 'ft',
-                'time': 's',
-                'duration': 's',
-                'initial': 's',
-                'EAS': 'kn',
-                'TAS': 'kn',
-                'angle': 'deg',
-                'pitch': 'deg',
-                'normal': 'lbf',
-                'final_alt': 'ft',
-                'required_available_climb_rate': 'ft/min',
-            }
-
-            if phase_name in ['accel', 'climb1', 'climb2', 'desc1', 'desc2']:
-                expected_units['distance'] = 'NM'
-
-            # loop through all trimmed_phase_options and call wrapped_convert_units with the correct expected units
-            for key, value in trimmed_phase_options.items():
-                for expected_key, expected_unit in expected_units.items():
-                    if key.startswith(expected_key):
-                        trimmed_phase_options[key] = wrapped_convert_units(
-                            value, expected_unit)
-
-            phase = phase_func(
-                ode_args=self.ode_args,
-                transcription=transcription,
-                **trimmed_phase_options)
-
+        else:
+            transcription = None
+
+        if 'groundroll' in phase_name:
+            phase_builder = GroundrollPhase
+        elif 'rotation' in phase_name:
+            phase_builder = RotationPhase
+        elif 'accel' in phase_name:
+            phase_builder = AccelPhase
+        elif 'ascent' in phase_name:
+            phase_builder = AscentPhase
+        elif 'climb' in phase_name:
+            phase_builder = ClimbPhase
+        elif 'cruise' in phase_name:
+            phase_builder = CruisePhase
+        elif 'desc' in phase_name:
+            phase_builder = DescentPhase
+
+        phase_object = phase_builder.from_phase_info(
+            phase_name, phase_options, default_mission_subsystems, meta_data=self.meta_data, transcription=transcription)
+        phase = phase_object.build_phase(aviary_options=self.aviary_inputs)
+
+        if 'cruise' not in phase_name:
```

Comment:
Just to clarify, "cruise" is still necessary to be in the name of a 2dof cruise phase, but it isn't necessary for height-energy?

Reply:
You got it!